node/src/components/block_synchronizer/global_state_synchronizer.rs

```diff
@@ -291,9 +288,10 @@ impl GlobalStateSynchronizer {
 
         self.in_flight.remove(&trie_hash);
         effect_builder
-            .put_trie_if_all_children_present(*trie_raw)
+            .put_trie_if_all_children_present((*trie_raw).clone())
             .event(move |put_trie_result| Event::PutTrieResult {
                 trie_hash,
+                trie_raw: *trie_raw,
```

Comment:
Ah, I see! Now we're cloning every time, instead of just in the error case. Then maybe it _is_ better to have the cloned value in the error? (Sorry! :grimacing:)

(Alternatively… where _does_ the `trie_raw` get dropped in the end? Is there a way to avoid the cloning altogether? Not a blocker, of course!)

Reply:
I guess it would be possible to avoid cloning if we passed it by value, instead of by reference, to `put_trie_if_all_children_present` and returned it back in the error variant - not sure if that's a reasonable API, though.